core/snapshot/config.go

```diff
@@ -39,16 +46,18 @@ func NewDefaultConfig() Config {
 		KeepRecent:  10,
 		RetryLimit:  5,
 		Storage:     LevelDB,
-		StartHeight: -1,
+		StartHeight: 0,
 	}
 }
 
 // Validate checks the values in the config file are sensible.
 func (c *Config) Validate() error {
-	// default to min 1 version, just so we don't have to account for nil slice.
-	// A single version kept in memory is pretty harmless.
-	if c.KeepRecent < 1 {
-		c.KeepRecent = 1
+	if c.KeepRecent < 1 || c.KeepRecent > 10 {
+		return ErrKeepRecentMustBeBetween1And10
+	}
+
+	if c.StartHeight < 0 {
```

Comment:
This is going to cause a lot of mayhem given that the system-tests/devops scripts etc will all be using `-1` at the moment. Maybe we can do:
```
if c.StartHeight < 0 {
    log.Warn("some warning to tell people to update their configs")
    c.StartHeight = 0
}
```
as a first step?

Reply:
I don't think anybody is monitoring the logs for such things. A more effective way would be to run some kind of migration / update script before or upon start up. But I don't think we have any of this, so it's very likely we are just going to delay the inevitable.

Reply:
Yea thats a fair point. I've kicked off a full system-test so we can see which tests will be affected:
https://jenkins.ops.vega.xyz/blue/organizations/jenkins/common%2Fsystem-tests/detail/system-tests/23783/pipeline

Reply:
Killing it and rerunning it based on latest changes.

https://jenkins.ops.vega.xyz/blue/organizations/jenkins/common%2Fsystem-tests/detail/system-tests/23792/pipeline